gpytorch/distributions/multivariate_normal.py

```diff
@@ -201,15 +202,32 @@ def to_data_independent_dist(self):
             import torch.distributions as base_distributions
         return base_distributions.Normal(self.mean, self.stddev)
 
+    @property
+    def stddev(self):
+        # self.variance is guaranteed to be positive, because we do clamping.
+        return self.variance.sqrt()
+
     @property
     def variance(self):
         if self.islazy:
             # overwrite this since torch MVN uses unbroadcasted_scale_tril for this
             diag = self.lazy_covariance_matrix.diag()
             diag = diag.view(diag.shape[:-1] + self._event_shape)
-            return diag.expand(self._batch_shape + self._event_shape)
+            variance = diag.expand(self._batch_shape + self._event_shape)
         else:
-            return super().variance
+            variance = super().variance
+
+        # Check to make sure that variance isn't lower than minimum allowed value (default 1e-6).
+        # This ensures that all variances are positive
+        min_variance = settings.min_variance.value()
+        if variance.lt(min_variance):
```

Comment:
should be `torch.any(variance.lt(min_variance))`, no?

Reply:
Oh yeah... good call.